packages/protocol/engine/tangle/blockdag/blockdag.go

```diff
@@ -139,6 +143,16 @@ func (b *BlockDAG) markSolid(block *Block) (err error) {
 	return nil
 }
 
+func (b *BlockDAG) checkTimestampMonotonicity(block *Block) error {
+	for _, parentID := range block.Parents() {
+		parent, parentExists := b.Block(parentID)
```

Comment:
What happens if the parent doesn't exist? I suppose it should never happen?

Reply:
If the parent doesn't exist then it means that it was part of a pruned epoch, and since the current block is being marked as solid, it's not part of a pruned epoch and therefore younger.